src/acvp.c

```diff
@@ -4796,36 +4795,27 @@ static ACVP_RESULT acvp_dispatch_vector_set (ACVP_CTX *ctx, JSON_Object *obj) {
         ACVP_LOG_ERR("JSON parse error: ACV algorithm not found");
         return ACVP_JSON_ERR;
     }
-
+    
     ACVP_LOG_STATUS("vsId: %d", vs_id);
     ACVP_LOG_STATUS("ACV Operation: %s", alg);
     // TODO: make sure this is included where relevant only
     ACVP_LOG_INFO("ACV Direction: %s", dir);
     
     ACVP_LOG_INFO("ACV version: %s", json_object_get_string(obj, "acvVersion"));
-
-    alg_tbl_index = calloc(12, sizeof(char));
     
-    if (strncmp(alg, "RSA", 3) == 0 || strncmp(alg, "ECDSA", 5) == 0) {
-        strncat((char *)alg_tbl_index, alg, 5);
-        strncat((char *)alg_tbl_index, "-", 1);
-        strncat((char *)alg_tbl_index, mode, 6);
-    } else {
-        strncpy((char *)alg_tbl_index, alg, strnlen(alg, 12));
-    }
-
     for (i = 0; i < ACVP_ALG_MAX; i++) {
-        if (!strncmp(alg_tbl_index, alg_tbl[i].name, strlen(alg_tbl[i].name))) {
-            rv = (alg_tbl[i].handler)(ctx, obj);
-            free((void *)alg_tbl_index);
-            return rv;
-        } else if (mode && !strncmp(mode, alg_tbl[i].name, strlen(alg_tbl[i].name))) {
-            rv = (alg_tbl[i].handler)(ctx, obj);
-            free((void *)alg_tbl_index);
-            return rv;
+        if (!strncmp(alg, alg_tbl[i].name, strlen(alg_tbl[i].name))) {
+            if (alg_tbl[i].mode != NULL) {
```

Comment:
May want to check to make sure mode is not null here before trying to do the strncm
That would cover the case where the server sent bad data, ie "RSA" but failed to send the mode such as "keyGen".

Reply:
Fixed :)